src/spectral/rulesets/ibm-oas/string-boundary.js

```diff
@@ -0,0 +1,70 @@
+module.exports = function(schema, _opts, paths) {
+  const rootPath = paths.target !== void 0 ? paths.target : paths.given;
+  return traverseSchema(schema, rootPath);
+};
+
+function traverseSchema(schema, path) {
+  if (schema.type === 'string') {
+    return stringBoundaryErrors(schema, path);
+  }
+  const errors = [];
+  if (schema.properties) {
+    Object.entries(schema.properties).forEach(function(prop) {
+      const propName = prop[0];
+      const propSchema = prop[1];
+      errors.push(
+        ...traverseSchema(propSchema, [...path, 'properties', propName])
+      );
+    });
+  } else if (schema.items) {
+    errors.push(...traverseSchema(schema.items, [...path, 'items']));
+  } else if (schema.allOf || schema.anyOf || schema.oneOf) {
+    const whichComposedSchemaType = schema.allOf
+      ? 'allOf'
+      : schema.anyOf
+      ? 'anyOf'
+      : 'oneOf';
+    const composedSchemas = schema[whichComposedSchemaType];
+    if (Array.isArray(composedSchemas)) {
+      composedSchemas.forEach(function(composedSchema, index) {
+        // TODO: is this the right way to get the index?
```

Comment:
Sure, that definitely works. There are other ways but they are only different, not necessarily better